src/plugins/flexibleLayout/components/frame.vue

```diff
@@ -0,0 +1,129 @@
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+<template>
+    <div class="c-fl-frame"
+        :class="{
+            'is-dragging': isDragging,
+            [frame.cssClass]: true
+        }">
+
+        <div class="c-frame c-fl-frame__drag-wrapper is-selectable is-moveable"
+             :class="{'no-frame': noFrame}"
+             draggable="true"
+             ref="frame"
+             v-if="frame.domainObject">
+
+             <frame-header 
+                v-if="index !== 0"
+                ref="dragObject"
+                :domainObject="frame.domainObject">
+            </frame-header>
+
+            <object-view
+                class="c-object-view"
+                :object="frame.domainObject">
+            </object-view>
+
+            <div class="c-fl-frame__size-indicator"
+                 v-if="frame.height && frame.height < 100 && isEditing">
+                {{frame.height}}%
+            </div>
+        </div>
+
+        <drop-hint
+             v-show="isEditing && isDragging"
+             class="c-fl-frame__drop-hint"
+             :class="{'is-dragging': isDragging}"
+             @object-drop-to="dropHandler">
+        </drop-hint>
+    </div>
+</template>
+
+<script>
+import ObjectView from '../../../ui/components/layout/ObjectView.vue';
+import DropHint from './dropHint.vue';
+import ResizeHandle from './resizeHandle.vue';
+import FrameHeader from '../../../ui/components/utils/frameHeader.vue';
+
+export default {
+    inject: ['openmct', 'domainObject'],
+    props: ['frame', 'index', 'containerIndex', 'size', 'isEditing', 'isDragging'],
+    data() {
+        return {
+            noFrame: this.frame.noFrame
+        }
+    },
+    components: {
+        ObjectView,
+        DropHint,
+        ResizeHandle,
+        FrameHeader
+    },
+    methods: {
+        dragstart(event) {
+            this.$emit('frame-drag-from', this.index);
+        },
+        dropHandler(event) {
+            this.$emit('frame-drop-to', this.index, event);
+        },
+        drag(event) {
+            if (!this.isDragging) {
+                this.isDragging = true;
+            }
+        },
+        deleteFrame() {
+            this.$emit('delete-frame', this.index);
+        },
+        addContainer() {
+            this.$emit('add-container');
+        },
+        toggleFrame(v) {
+            this.noFrame = v;
+        }
+    },
+    mounted() {
+        this.$el.addEventListener('dragstart', this.dragstart);
+        this.$el.addEventListener('drag', this.drag);
+
+        if (this.frame.domainObject.identifier) {
+                let context = {
+                item: this.frame.domainObject,
+                parentDomainObject: this.domainObject,
+                method: this.deleteFrame,
+                addContainer: this.addContainer,
```

Comment:
this context is huge-- you're duplicating information that you could get from selection context. check toolbar note for how to improve.